### ISSUE TITLE
Linking FirebaseInstallation fails with lld

### ISSUE THREAD

**Issue by sebastianv1** [firebase/firebase-ios-sdk#10071]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 13.4.1
  * Firebase SDK version: 8.7.0
  * Installation method: `Zip file` (select one)
  * Firebase Component: Analytics
  * Target platform(s): `All`

### [REQUIRED] Step 2: Describe the problem
When linking our app using `lld`, we receive the following error: 
```
error: framework not found for -framework FBLPromises
```

We discovered that this is coming from `FirebaseInstallation` whose `LC_LINKER_OPTIONS` includes `FBLPromises` as a load command
```
...
  string #1 -framework
  string #2 FBLPromises
Load command 5
     cmd LC_LINKER_OPTION
...
```

It appears that `FBLPromises` was renamed to `PromisesObjC` and is a dependency of `FirebaseInstallation` & included in the zipped release of frameworks. We do link `PromisesObjC` into our bundle, but my guess is that the distribution step still links against the old `FBLPromises` name?

This doesn't throw an error in `ld` though since it'll ignore invalid frameworks included by the `LC_LINKER_OPTIONS` flag (https://github.com/llvm/llvm-project/issues/56939).
#### Steps to reproduce:
Link an app using `lld` against `FirebaseInstallation` + dependencies.

What happened? How can we make the problem occur?
`error: framework not found for -framework FBLPromises`

If you have a downloadable sample project that reproduces the bug you're reporting, you will
likely receive a faster response on your issue.

**Comment by paulb777** [comment]
Thanks for the report. This may be a CocoaPods issue with the Promises dependency.  How serious is the issue for you? Is it more than an annoyance in the build log?

**Comment by sebastianv1** [comment]
@paulb777 The issue is an error using `lld`, not just an annoyance in the build log. `ld` still works, but it'd be awesome to use `lld` since we're seeing >50% improvements in linking times.

Does the zipped distribution get built from the podspecs? Looks like the PromisesObjC `module_name` is set to `FBLPromises` which could be the source of the issue? (https://github.com/google/promises/blob/master/PromisesObjC.podspec#L20)

**Comment by paulb777** [comment]
Yes, the zip distribution is created from the podspecs. It looks like our build tooling is trying to sort out the module name/pod name conflation at https://github.com/firebase/firebase-ios-sdk/blob/master/ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift#L291

cc: @ykjchen

**Comment by paulb777** [comment]
@sebastianv1 Does renaming `PromisesObjC.xcframework` to `FBLPromises.xcframework` make a difference?

**Comment by sebastianv1** [comment]
@paulb777 I have not tried this yet, currently we are able to workaround it by using an empty FBLPromises dummy framework that we include the link list. Would it help if I tried?

**Comment by paulb777** [comment]
@sebastianv1 Yes. It's essentially what the fix in #10073 does and I'd like to confirm it works for your use case.

**Comment by paulb777** [comment]
I believe that #10073 should fix the issue and it will be part of the next 9.5.0 release.
